java-frontend/src/main/java/org/sonar/java/regex/ast/IndexRange.java

```diff
@@ -0,0 +1,67 @@
+/*
+ * SonarQube Java
+ * Copyright (C) 2012-2020 SonarSource SA
+ * mailto:info AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+package org.sonar.java.regex.ast;
+
+import java.util.Objects;
+
+public class IndexRange {
+
+  private final int beginningOffset;
+  private final int endingOffset;
+
+  public IndexRange(int beginningOffset, int endingOffset) {
+    this.beginningOffset = beginningOffset;
+    this.endingOffset = endingOffset;
+  }
+
+  public int getBeginningOffset() {
+    return beginningOffset;
+  }
+
+  public int getEndingOffset() {
+    return endingOffset;
+  }
+
+  public IndexRange merge(IndexRange other) {
+    return new IndexRange(beginningOffset, other.endingOffset);
```

Comment:
This merging can only work if the IndexRange given as argument is AFTER this IndexRange. Could it be done the other way or by construction we will always call it correctly?

Reply:
Barring bugs in the parser, it should only ever be called with the earlier index range first. It might be a good idea to verify this and throw an InvalidArgumentException otherwise though.